src/openfl/display/DisplayObject.hx

```diff
@@ -1325,26 +1320,92 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 	}
 	
 	
-	public function __updateChildren (transformOnly:Bool):Void {
+	private function __updateRenderData ():Void {
 		
 		var renderParent = __renderParent != null ? __renderParent : parent;
+		if (__isMask && renderParent == null) renderParent = __maskTarget;
 		__renderable = (visible && __scaleX != 0 && __scaleY != 0 && !__isMask && (renderParent == null || !renderParent.__isMask));
-		__worldAlpha = alpha;
-		__worldBlendMode = blendMode;
 		
-		if (__transformDirty) {
+		__updateTransforms ();
+		
+		if (__supportDOM) {
+				
+				__renderTransformChanged = !__renderTransform.equals (__renderTransformCache);
```

Comment:
I think the spacing from here is not right. I think all this needs to go one tab to the left right?